Start-C4bJenkinsSetup.ps1

```diff
@@ -0,0 +1,181 @@
+    'token-macro' = '2.13'
+    'build-timeout' = '1.20'
+    'credentials' = '2.5'
+    'plain-credentials' = '1.7'
+    'ssh-credentials' = '1.18.1'
+    'credentials-binding' = '1.24'
+    'scm-api' = '2.6.4'
+    'workflow-api' = '2.46'
+    'timestamper' = '1.13'
+    'caffeine-api' = '2.9.1-23.v51c4e2c879c8'
+    'script-security' = '1.77'
+    'plugin-util-api' = '1.7.1'
+    'font-awesome-api' = '5.15.2-1'
+    'popper-api' = '1.16.1-1'
+    'jquery3-api' = '3.5.1-2'
+    'bootstrap4-api' = '4.6.0-1'
+    'snakeyaml-api' = '1.29.1'
+    'jackson2-api' = '2.12.3'
+    'echarts-api' = '4.9.0-3'
+    'display-url-api' = '2.3.5'
+    'workflow-support' = '3.8'
+    'workflow-job' = '2.41'
+    'checks-api' = '1.5.0'
+    'junit' = '1.50'
+    'matrix-project' = '1.18'
+    'resource-disposer' = '0.16'
+    'ws-cleanup' = '0.39'
+    'ant' = '1.11'
+    'durable-task' = '1.37'
+    'workflow-durable-task-step' = '2.35'
+    'command-launcher' = '1.6'
+    'jdk-tool' = '1.5'
+    'bouncycastle-api' = '2.20'
+    'ace-editor' = '1.1'
+    'workflow-scm-step' = '2.13'
+    'workflow-cps' = '2.92'
+    'apache-httpcomponents-client-4-api' = '4.5.13-1.0'
+    'mailer' = '1.32.1'
+    'workflow-basic-steps' = '2.21'
+    'gradle' = '1.36'
+    'pipeline-milestone-step' = '1.3.2'
+    'pipeline-input-step' = '2.12'
+    'pipeline-stage-step' = '2.5'
+    'pipeline-graph-analysis' = '1.11'
+    'pipeline-rest-api' = '2.19'
+    'handlebars' = '3.0.8'
+    'momentjs' = '1.1.1'
+    'pipeline-stage-view' = '2.19'
+    'pipeline-build-step' = '2.13'
+    'pipeline-model-api' = '1.8.5'
+    'pipeline-model-extensions' = '1.8.5'
+    'jsch' = '0.1.55.2'
+    'git-client' = '3.6.0'
+    'git-server' = '1.9'
+    'workflow-cps-global-lib' = '2.19'
+    'branch-api' = '2.6.2'
+    'workflow-multibranch' = '2.24'
+    'pipeline-stage-tags-metadata' = '1.8.5'
+    'pipeline-model-definition' = '1.8.5'
+    'lockable-resources' = '2.11'
+    'workflow-aggregator' = '2.6'
+    'jjwt-api' = '0.11.2-9.c8b45b8bb173'
+    'okhttp-api' = '3.14.9'
+    'github-api' = '1.123'
+    'git' = '4.6.0'
+    'github' = '1.33.1'
+    'github-branch-source' = '2.9.9'
+    'pipeline-github-lib' = '1.0'
+    'mapdb-api' = '1.0.9.0'
+    'subversion' = '2.14.4'
+    'ssh-slaves' = '1.31.5'
+    'matrix-auth' = '2.6.7'
+    'pam-auth' = '1.6'
+    'ldap' = '1.25'
+    'email-ext' = '2.83'
+    'powershell' = '1.5'
+}
+
+# Performance is killed by Invoke-WebRequest's progress bars, turning them off to speed this up
+$ProgressPreference = 'SilentlyContinue'
+
+foreach ($PluginName in $JenkinsPlugins.Keys) {
+    $PluginUri = 'https://updates.jenkins-ci.org/download/plugins/{0}/{1}/{0}.hpi' -f $PluginName, $JenkinsPlugins[$PluginName]
+    $PluginPath = '{0}/plugins/{1}.hpi' -f $jenkinsHome, $PluginName
+
+    Invoke-WebRequest -Uri $PluginUri -OutFile $PluginPath -UseBasicParsing
+}
+
+# Restore default progress bar setting
+$ProgressPreference = 'Continue'
```

Comment:
We could avoid IWR entirely here, and be faster
```suggestion
foreach($PluginName in $JenkinsPlugins.Keys) {
  $PluginUri = 'https://updates.jenkins-ci.org/download/plugins/{0}/{1}/{0}.hpi' -f $PluginName,$JenkinsPlugins[$PluginName]
  $PluginPath = '{0}/plugins/{1}.hpi -f $JenkinsHome, $PluginName
  
  [System.Net.WebClient]::New().DownloadFile($PluginUri,$PluginPath)
```

Reply:
This is a good idea. Two things:
- This is straight from [code](https://github.com/chocolatey/qde-azure/blob/main/application/source/Install-Jenkins.ps1) by @JPRuskin and @Windos, so just tagging them for awareness
- We can probably get rid of the Progress preference then too